### DISCUSSION TITLE
Brittle command args options locations

### DISCUSSION THREAD

**Discussion by nextmat** [erikhuda/thor#846] [Ideas]
I'm noticing in using thor that it sees to be pretty rigid about option/flag locations and argument locations. For example:

```
mycli touch <file> --opt   # valid
mycli touch --opt <file>   # not valid
```

Since a lot of unix CLI tools expect flags first, this may be confusing for users. Is there an existing way to modify this behavior? If not I'd be happy to contribute a patch for additional parsing to allow more flexibility in flag/argument locations.

**Comment by josevalim** [comment]
I don't remind on top of my mind if there is a configuration option for that. But keep in mind that the rule to have options at the end is important to be able to invoke a thor script from another one.

**Comment by nextmat** [comment]
Why is that? Can you give me an example?

**Comment by ctdean** [comment]
This is surprising behavior to me as well.  Love to see it change to allow an option before the command arguments

**Comment by nextmat** [comment]
Still planning to write a patch for this, just want to understand the current behavior better first to make sure I don't inadvertently break something.
